src/pool/pool_disjoint.c

```diff
@@ -657,6 +674,24 @@ umf_result_t disjoint_pool_initialize(umf_memory_provider_handle_t provider,
     *ppPool = (void *)disjoint_pool;
 
     return UMF_RESULT_SUCCESS;
+
+err_buckets_alloc_fail:
+    for (i = 0; i < disjoint_pool->buckets_num; i++) {
+        if (disjoint_pool->buckets[i] != NULL) {
+            destroy_bucket(disjoint_pool->buckets[i]);
+        }
+    }
+
+err_buckets_array_alloc_fail:
+    umf_ba_global_free(disjoint_pool->buckets);
```

Comment:
You should destroy `disjoint_pool->default_shared_limits` here.

Reply:
thanks, now it should be correct

Reply:
It is OK now.